### PR TITLE
Feature/138 vuex module decorators

### PR DESCRIPTION
**Pull request by TE4-Mattis-Abrahamsson**
## What issue did you implement or fix?
Refactored Vuex modules, issue #138 

## Summary
- Rewrote the module for books with vuex-module-decorators and renamed it to BooksModule
- Rewrote the module for titles with vuex-module-decorators and renamed it to TitlesModule
- Rewrote the module for loans with vuex-module-decorators and renamed it to LoansModule
- Rewrote the module for users with vuex-module-decorators and renamed it to UsersModule
- Switched from the old library for persisting state to vuex-persist
- Updated all files that were affected by above changes to use the new modules

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- There are warnings in the console, e.g. "[vuex] state field "TitlesModule" was overridden by a module with the same name at "TitlesModule"" these do not break anything but could possibly be fixed.

### CODE REVIEW

**Comment by TE4-Mattis-Abrahamsson** [comment]
The unit-test failed because of the new depedency (@bartvanvliet/vuex-module-decorators), not sure why or how this can be fixed. Ping @itggot-linus-styren

**Comment by itggot-linus-styren** [comment]
> The unit-test failed because of the new depedency (@bartvanvliet/vuex-module-decorators), not sure why or how this can be fixed. Ping @itggot-linus-styren

It seems the dependency is not present in package.json so how could the CI know it should be installed? Also, I think it is better to depend on vuex-module-decorators and then simply use [this patch](https://github.com/itggot-TE4/Log-O-Matic-WHIJO/blob/develop/patches/vuex-module-decorators%2B0.11.0.patch) with the `npx patch-package` command (add it is a step to the CI job).

**Comment by itggot-daniel-berg** [comment]
> It seems the dependency is not present in package.json so how could the CI know it should be installed? Also, I think it is better to depend on vuex-module-decorators and then simply use [this patch](https://github.com/itggot-TE4/Log-O-Matic-WHIJO/blob/develop/patches/vuex-module-decorators%2B0.11.0.patch) with the `npx patch-package` command (add it is a step to the CI job).

Another possibility might be to add  bartvanvliets fork to package.json and install it from there: https://github.com/Kapulara/vuex-module-decorators (supposing the fork works).

File: frontend/src/store/modules/LoansModule.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  loaned_by_id: number;
+  returned_at: string|null;
+  updated_at: string;
+}
+export interface LoanCollection {[id: number]: Loan; }
+
+export interface LoanState {
+  loans: LoanCollection;
+  failure: any;
+}
```
Comment (TE4-Daniel-Kull) [inline frontend/src/store/modules/LoansModule.ts]: Jag tror det kan vara en bra idé att flytta ut dessa i en egen fil, men jag är inte säker. Vad tycker du?

Comment (TE4-Mattis-Abrahamsson) [reply frontend/src/store/modules/LoansModule.ts]: Not sure, it could be a good idea, not sure what the recommended practice is for Typescript interfaces.

Comment (itggot-linus-styren) [reply frontend/src/store/modules/LoansModule.ts]: You could compare how the typescript team does it. See related: https://stackoverflow.com/q/36633033/3013334

**Pull request merged**